cmake/modules/extensions.cmake

```diff
@@ -2421,6 +2421,11 @@ Relative paths are only allowed with `-D${ARGV1}=<path>`")
                         BUILD ${FILE_BUILD}
     )
     list(APPEND filename_list ${filename})
+
+    foreach(shield ${SHIELD_AS_LIST})
+      list(APPEND filename_list ${shield})
+    endforeach()
```

Comment:
Copied from: https://github.com/zephyrproject-rtos/zephyr/pull/67204#discussion_r1447104359

this is quite a big change and impacts not only overlay's but also Kconfig fragments (and technical provide an unexpected feature for defconfig in HWMv2, ref: #67322)

Would like to open a discussion on the feature itself, also in relation to #67227, because with this commit proposal then applying two shields, where there might only be a sample shield overlay for first board, then that could result in surprises.

Regardless of above, some minimum changes are required on shield usage.
The `zephyr_file()` takes board as argument, and only defaults to system wide board when `BOARD` is not passed as argument: https://github.com/zephyrproject-rtos/zephyr/blob/b98d2a1b4744939a8a3259f112da90b3e4c648f7/cmake/modules/extensions.cmake#L2404-L2405

so we should as minimum ensure:
- [ ] Allow SHIELD list be provided as argument
- [ ] Don't use system wide SHIELD when BOARD is passed as argument
- [ ] Document extended behavior for conf files, https://github.com/zephyrproject-rtos/zephyr/blob/b98d2a1b4744939a8a3259f112da90b3e4c648f7/doc/build/kconfig/setting.rst#the-initial-configuration.